web/src/filter.ts

```diff
@@ -223,6 +223,21 @@ function message_matches_search_term(message: Message, operator: string, operand
             }
             return operand_ids.every((operand_id) => user_ids.includes(operand_id));
         }
+
+        case "mentions": {
+            const operand_user = people.get_by_email(operand);
+
+            if (!operand_user) {
+                return false;
+            }
+
+            const user_id_str = `data-user-id="${operand_user.user_id}"`;
+
+            return (
+                message.content.includes(user_id_str) &&
+                !message.content.includes("user-mention silent")
+            );
```

Comment:
This hacky check is not correct. Don't do hacky imprecise checks. You can parse the HTML instead, a la how `post_process_content` works, to find mention elements and inspect their classes and attributes directly.